week2/starter/ColorPicker/ContentView.swift

```diff
@@ -33,54 +33,82 @@
 import SwiftUI
 
 struct ContentView: View {
+  private let initialColor = Color(
+    red: Constants.Color.redInit / Constants.maxValue,
+    green: Constants.Color.greenInit / Constants.maxValue,
+    blue: Constants.Color.greenInit / Constants.maxValue
+  )
+  @Environment(\.verticalSizeClass) var verticalSizeClass
   @State private var alertIsVisible: Bool = false
-  @State private var redColor: Double = 0.0
-  @State private var greenColor: Double = 0.0
-  @State private var blueColor: Double = 0.0
-  @State private var foregroundColor = Color(red: 0, green: 0, blue: 0)
-
+  @State private var redColor: Double = Constants.Color.redInit
+  @State private var greenColor: Double = Constants.Color.greenInit
+  @State private var blueColor: Double = Constants.Color.blueInit
+  @State private var foregroundColor: Color
+  
+  init() {
+    _foregroundColor = State(initialValue: self.initialColor)
+  }
+  
   var body: some View {
-
-    VStack {
-      Text("Color Picker")
-        .font(.largeTitle)
-
-      RoundedRectangle(cornerRadius: 0)
-        .foregroundColor(foregroundColor)
-        .border(.black)
-      VStack {
-        Text("Red")
-        HStack {
-          Slider(value: $redColor, in: 0...255)
-          Text("\(Int(redColor.rounded()))")
-        }
-      }
-      VStack {
-        Text("Green")
-        HStack {
-          Slider(value: $greenColor, in: 0...255)
-          Text("\(Int(greenColor.rounded()))")
-        }
+    var ColorPickerApp = ColorPickerAppView(
+      foregroundColor: $foregroundColor,
+      redColor: $redColor,
+      greenColor: $greenColor,
+      blueColor: $blueColor
+    )
+    
+    // This is really buggy in Preview View, but works as expected in Simulator?!
+    // Will explore further...
+    if verticalSizeClass == .compact {
```

Comment:
Did you try using both horizontal and vertical size classes? Maybe that would get rid of the bug you mentioned. I did: horizontalSizeClass == .compact && verticalSizeClass == .regular and experienced no issues.